src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -314,5 +386,31 @@ async def debug_python_test(field_set: PythonTestFieldSet) -> TestDebugRequest:
     )
 
 
+# -----------------------------------------------------------------------------------------
+# `runtime_package_dependencies` plugin
+# -----------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class RuntimePackagesPluginRequest(PytestPluginSetupRequest):
+    @classmethod
+    def is_applicable(cls, target: Target) -> bool:
+        return bool(target.get(RuntimePackageDependenciesField).value)
+
+
+@rule
+async def setup_runtime_packages(request: RuntimePackagesPluginRequest) -> PytestPluginSetup:
+    built_packages = await Get(
+        BuiltPackageDependencies,
+        BuildPackageDependenciesRequest(request.target.get(RuntimePackageDependenciesField)),
+    )
```

Comment:
With this here, shouldn't you also remove this from inside `setup_pytest_for_target`? Aren't they duplicating this logic?
```
    build_package_dependencies_get = Get(
        BuiltPackageDependencies,
        BuildPackageDependenciesRequest(request.field_set.runtime_package_dependencies),
    )
    ...
    (
        ...
        built_package_dependencies,
        ...
    ) = await MultiGet(
        ...
        build_package_dependencies_get,
        ...
    )
    input_digest = await Get(
        Digest,
        MergeDigests(
            (
                ...
                *(pkg.digest for pkg in built_package_dependencies),
```

Reply:
Indeed, thank you for catching that! 🙏